keystore/keystore.go

```diff
@@ -149,5 +153,19 @@ func (ks *FSKeystore) List() ([]string, error) {
 		return nil, err
 	}
 
-	return dir.Readdirnames(0)
+	dirs, err := dir.Readdirnames(0)
+	if err != nil {
+		return nil, err
+	}
+
+	var list []string
+
+	for _, name := range dirs {
+		err := validateName(name)
+		if err == nil {
+			list = append(list, name)
+		}
+	}
+
+	return list, err
```

Comment:
Returning the error here is a bit confusing. It should always be nil (the go idiom is to just return `list, nil`).